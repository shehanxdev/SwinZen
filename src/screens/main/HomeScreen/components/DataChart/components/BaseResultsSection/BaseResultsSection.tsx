import React from 'react';
import { View } from 'react-native';

import { tw } from '@sz/config';

import { BaseChartBar } from '../BaseChartBar/BaseChartBar';

interface BaseResultsSectionProps {
  passes: number;
  fails: number;
}

export function BaseResultsSection({ passes, fails }: BaseResultsSectionProps) {
  return (
    <View style={tw`flex flex-row justify-start`}>
      <BaseChartBar barValue={passes} chartBarType={'pass'} />
      <BaseChartBar barValue={fails} chartBarType={'fail'} />
    </View>
  );
}
