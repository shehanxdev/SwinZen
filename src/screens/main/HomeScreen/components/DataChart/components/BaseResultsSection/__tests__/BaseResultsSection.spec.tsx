import { render } from '@testing-library/react-native';
import * as React from 'react';

import { BaseResultsSection } from '../BaseResultsSection';

describe('BaseResultsSection component', () => {
  const propsSet = [
    { passes: 10, fails: 5, label: 'Overall' },
    { passes: 8, fails: 2, label: 'Setup' },
    { passes: 1, fails: 3, label: 'Backswing' },
    { passes: 2, fails: 5, label: 'Downswing' },
  ];

  describe('should render correctly', () => {
    for (const prop of propsSet) {
      it(`should render correctly with ${JSON.stringify(prop)}`, () => {
        const rendered = render(<BaseResultsSection {...prop} />);
        const renderedTree = rendered.toJSON();
        expect(renderedTree).toMatchSnapshot();
      });
    }
  });
});
