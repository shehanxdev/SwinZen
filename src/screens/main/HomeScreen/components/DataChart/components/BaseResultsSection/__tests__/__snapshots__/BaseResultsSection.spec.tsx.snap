// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseResultsSection component should render correctly should render correctly with {"passes":1,"fails":3,"label":"Backswing"} 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4288871727,
            1419967791,
            10681647,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 23.5,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#A2FD2F",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {},
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#16501D",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4294334485,
            1307990037,
            16144405,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 70.5,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#F65815",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3,
            },
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          3
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": -48,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 15,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      Backswing
    </Text>
  </View>
</View>
`;

exports[`BaseResultsSection component should render correctly should render correctly with {"passes":2,"fails":5,"label":"Downswing"} 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4288871727,
            1419967791,
            10681647,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 47,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#A2FD2F",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {},
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#16501D",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4294334485,
            1307990037,
            16144405,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 117.5,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#F65815",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3,
            },
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          5
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": -48,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 15,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      Downswing
    </Text>
  </View>
</View>
`;

exports[`BaseResultsSection component should render correctly should render correctly with {"passes":8,"fails":2,"label":"Setup"} 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4288871727,
            1419967791,
            10681647,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 188,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#A2FD2F",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {},
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#16501D",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          8
        </Text>
      </View>
    </View>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4294334485,
            1307990037,
            16144405,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 47,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#F65815",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3,
            },
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": -48,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 15,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      Setup
    </Text>
  </View>
</View>
`;

exports[`BaseResultsSection component should render correctly should render correctly with {"passes":10,"fails":5,"label":"Overall"} 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4288871727,
            1419967791,
            10681647,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 235,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#A2FD2F",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {},
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#16501D",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          10
        </Text>
      </View>
    </View>
  </View>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
    onLayout={[Function]}
    style={
      {
        "alignItems": "flex-end",
        "justifyContent": "flex-end",
      }
    }
    testID="BaseChartBarComponentTestID"
  >
    <View>
      <BVLinearGradient
        colors={
          [
            4294334485,
            1307990037,
            16144405,
          ]
        }
        endPoint={
          {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          [
            0,
            0.48,
            1,
          ]
        }
        startPoint={
          {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          {
            "height": 117.5,
            "width": 36,
          }
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#F65815",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "absolute",
            },
            {
              "shadowColor": "#000000",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3,
            },
            {
              "height": 36,
              "top": -18,
              "width": 36,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          5
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": -48,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 15,
          },
          {
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      Overall
    </Text>
  </View>
</View>
`;
